### PR TITLE
generate rss for all MIT podcast episodes

### PR DESCRIPTION
**Pull request by abeglova**
#### Pre-Flight checklist
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
closes https://github.com/mitodl/open-discussions/issues/3047

#### What's this PR do?
This pr updates the podcasts pipeline to generate an rss file with all MIT episodes.
It also creates a view to display the rss file at /podcasts/rss_feed

#### How should this be manually tested?
run `docker-compose run web ./manage.py backpopulate_podcast_data`
A rss file should be created in the `/podcast_rss directory`
http://localhost:8063/podcasts/rss_feed should display the rss feed

### CODE REVIEW

File: course_catalog/etl/podcast.py [hunk 1]
```diff
@@ -199,3 +200,73 @@ def transform(extracted_podcasts):
...
+    Returns:
+        Beautiful soup object of the rss for the  MIT aggregate podcast, excluding episodes
+    """
+    current_timestamp = datetime.now().strftime(TIMESTAMP_FORMAT)
+
+    rss = """<?xml version='1.0' encoding='UTF-8'?>
+    <rss xmlns:itunes="http://www.itunes.com/dtds/podcast-1.0.dtd" version="2.0">
+        <channel>
+            <title>MIT Open Aggregated Podcast Feed</title>
+            <link>https://open.mit.edu/podcasts/<link>
```
Comment (noisecapella) [inline course_catalog/etl/podcast.py:217]: Can we generate this from `SITE_BASE_URL`?

File: course_catalog/etl/podcast.py [hunk 2]
```diff
@@ -199,3 +200,73 @@ def transform(extracted_podcasts):
...
+            <pubDate> {0} </pubDate>
+            <lastBuildDate> {0} </lastBuildDate>
+            <ttl>60</ttl>
+            <itunes:subtitle>Episodes from podcasts from around MIT</itunes:subtitle>
+            <itunes:author>MIT Open Learning</itunes:author>
+            <itunes:summary>Episodes from podcasts from around MIT</itunes:summary>
+            <description>Episodes from podcasts from around MIT</description>
+            <itunes:owner>
+                <itunes:name>MIT Open Learning</itunes:name>
+                <itunes:email>odl-discussions-support@mit.edu</itunes:email>
```
Comment (noisecapella) [inline course_catalog/etl/podcast.py:228]: We should use `settings.EMAIL_SUPPORT` here

File: course_catalog/etl/podcast.py [hunk 3]
```diff
@@ -199,3 +200,73 @@ def transform(extracted_podcasts):
...
+    episode_rss_list = sorted(
+        episode_rss_list,
+        key=lambda episode: datetime.strptime(episode.pubDate.text, TIMESTAMP_FORMAT),
+        reverse=True,
+    )
+
+    for episode in episode_rss_list:
+        channel_rss.channel.append(episode)
+
+    with open(RSS_FILE_LOCATION, "w") as file:
```
Comment (noisecapella) [inline course_catalog/etl/podcast.py:258]: I don't think we should be writing to our own file system. For one, anything written will be erased on the next restart of the heroku server. Instead, we could store the relevant parts of the RSS in the database in a `PodcastEpisode` field or a field on a related model, then generate the XML from that when the user views `podcast_rss_feed`.

**Comment by abeglova** [comment]
Thanks @noisecapella. Back to you

**Pull request merged**